react-components/src/library/components/search/auto-suggest.js

```diff
@@ -0,0 +1,226 @@
+          const suggestion = suggestions[selectedSuggestionIndex]
+          if (suggestion !== undefined) {
+            this.setState({ query: suggestion, showSuggestions: false, selectedSuggestionIndex: -1 }, () => {
+              const { onChange, onSubmit } = this.props
+              if (onChange) {
+                onChange(suggestion)
+              }
+              if (onSubmit) {
+                // allow onChange to settle in stem-finder
+                setTimeout(() => onSubmit(), 1)
+              }
+            })
+            handledKey = true
+          }
+        }
+        break
+      case 27: // escape
+        if (showSuggestions) {
+          this.setState({ showSuggestions: false, selectedSuggestionIndex: -1 })
+          handledKey = true
+        }
+        break
+      case 38: // up arrow
+        if (showSuggestions) {
+          if (selectedSuggestionIndex > 0) {
+            this.setState({ selectedSuggestionIndex: selectedSuggestionIndex - 1 })
+          } else {
+            this.setState({ showSuggestions: false })
+          }
+          handledKey = true
+        }
+        break
+      case 40: // down arrow
+        if (showSuggestions) {
+          if (selectedSuggestionIndex < suggestions.length - 1) {
+            this.setState({ selectedSuggestionIndex: selectedSuggestionIndex + 1 })
+            handledKey = true
+          }
+        } else {
+          this.setState({ showSuggestions: true })
+          handledKey = true
+        }
+        break
+    }
+
+    if (handledKey) {
+      e.preventDefault()
+      e.stopPropagation()
+    }
+  }
+
+  renderSuggestions () {
+    const { suggestions, selectedSuggestionIndex, showSuggestions } = this.state
+
+    if (!showSuggestions || (suggestions.length === 0)) {
+      return undefined
+    }
+
+    const items = suggestions.map((suggestion, index) => {
+      const selected = index === selectedSuggestionIndex
+      return <Suggestion key={suggestion} suggestion={suggestion} selected={selected} onClick={this.handleSuggestionClick} />
+    })
+
+    let style = {}
+    if (this.inputRef.current) {
+      const width = this.inputRef.current.getBoundingClientRect().width
+      style = { width }
+    }
+
+    return (
+      <div id='suggestions' className={css.suggestions} style={style}>
+        { items }
+      </div>
+    )
+  }
+
+  render () {
+    const { name, placeholder, id } = this.props
+
+    return (
+      <div className={css.autoSuggest} ref={this.containerRef}>
+        <input
+          id={id || undefined}
+          ref={this.inputRef}
+          name={name || undefined}
+          placeholder={placeholder}
+          type='text'
+          autoComplete='off'
+          value={this.state.query}
+          onChange={this.handleInputChange}
+          onKeyDown={this.handleKeyDown}
+        />
+        {this.renderSuggestions()}
+      </div>
+    )
+  }
+}
+
+// TODO: add handler to hide suggestions on click outside of suggestions
```

Comment:
It looks like this is taken care of, at least the suggestions were hidden when I tested locally.

Reply:
Yep - it is handled now.  I was using the bottom of this file to keep track of what to do :)  I've removed in this commit: https://github.com/concord-consortium/rigse/pull/810/commits/a624a7f2f341d65e09854d8c94c9ca9448dd9646